specification/compute/resource-manager/Microsoft.Compute/preview/2016-04-30-preview/compute.json

```diff
@@ -159,6 +159,34 @@
           }
         }
       }
+    },
+	"/subscriptions/{subscriptionId}/providers/Microsoft.Compute/availabilitySets": {
+      "get": {
+        "tags": [
+          "AvailabilitySets"
+        ],
+        "operationId": "AvailabilitySets_ListBySubscription",
+        "description": "Lists all availability sets in a subscription.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/AvailabilitySetListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
```

Comment:
The provided example creates a linter error:
> The property 'nextLink' specified by nextLinkName does not exist in the 200 response schema. \nPlease, specify the name of the property that provides the nextLink. If the model does not have the nextLink property then specify null.

Could you either:
- Update the example to include an example of nextLink if server supports it
- Update the Swagger to replace "nextLink" by null if server does not support nextLink

Whatever you do, do NOT remove "x-ms-pageable", that's the right thing to do since it's a method that has the semantic of a list.

Reply:
Can you please check again? I think the issue is not that nextLink is missing in the example, but that nextLink is missing AvailabilitySetListResult in compute.json.

Reply:
Ok, I see what you mean, you re-used `AvailabilitySetListResult` as it was defined. Let me check what should be the correct way to remove this warning then.

Reply:
@dochung4 I made an actual call to the endpoint, and didn't get a nextLink back in the JSON. Based  on that, correct definition is to use `null`.
Could you confirm the actual server behavior:
- If server never returns nextLink, use `null`
- If server may return nextLink, use what you did `nextLink`

Swagger must described accurately the server, and since I don't own the server implementation I don't know which one is the truth. Your last commit changed a lot of `null` to `nextLink` and that concerns me a bit.

Reply:
For which call did you not get a nextLink back? Currently we have nextLink for listing all availability sets in a subscription, but not for listing all resource groups in a resource group. But I discussed this with my team and I added nextLink to the latter too, because it will be enabled very soon.

Please let me know if you have other thoughts.

Reply:
I checked listing AS in a RG indeed. So if I understand you correctly:
- List AS at the subscription level always returns nextLink
- List AS at the RG level didn't have nextLink, but you're adding it live on all api-versions from 2016-04-30-preview to now soon.

Is this accurate?

Reply:
Yes that's correct.